config/src/lib.rs

```diff
@@ -180,49 +299,112 @@ mod tests {
             takes_argument: false,
             help: "a switch that can be provided - or not",
             documentation: "test doc switch",
+            list: false,
         };
+        pub const TEST_MULTIPLE: ConfigOption = ConfigOption {
+            name: "testmultiple",
+            default: "",
+            required: false,
+            takes_argument: true,
+            help: "A parameter that can be specified multiple times and all values will be used.",
+            documentation: "",
+            list: true,
+        };
+
+        // This function retrieves a string value that is stored for the ConfigOption that
+        // is passed to it
+        // The function will only return this value if it is the only value that is stored under
+        // this ConfigOption, it will panic in any other case:
+        //  - no value is stored
+        //  - more than one value is stored
+        // This allows keeping assert statements a bit briefer in the actual test by removing the
+        // need to check for empty or lists larger than one
+        pub fn get_first_and_only_value(&self, key: &ConfigOption) -> String {
+            let value = self
+                .values
+                .get(key)
+                .expect("Error retrieving value!")
+                .clone();
+            if value == None {
+                panic!("Argument was not specified!");
+            }
+            let value = value.expect("Shouldn't happen");
+            if value.len() != 1 {
+                panic!("Not a single value: {}", value.len());
+            }
+            String::from(&value[0].clone())
+        }
+
+        // Helper function to check whether the argument was provided on the command line
+        pub fn argument_was_provided(&self, key: &ConfigOption) -> bool {
+            let test = self.values.get(key).unwrap();
+
+            match self
+                .values
+                .get(key)
+                .expect("Fatal error: key not present in HashMap, but should have been!")
+            {
+                None => false,
+                Some(v) => true,
```

Comment:
Could we replace the Some(v) with the wildcard "_"?
`_ => true,`

Reply:
Indeed we could. I've rewritten the entire statement a little as an "if let" block instead of match, which seems to be the recommended way - even though I'm not sure I agree that it is more readable :)